app/models/course/assessment/question/programming.rb

```diff
@@ -62,15 +62,18 @@ def copy_template_files_to(answer)
   # We restore the original package, but capture the new package into a local for processing by
   # the import job.
   def process_new_package
-    return remove_old_package if attachment.nil?
+    return remove_old_package if attachment_reference.nil?
 
-    new_attachment = attachment
-    restore_attribute!(:attachment)
+    new_attachment_reference = attachment_reference
+    restore_attribute!(:attachment_reference)
 
     execute_after_commit do
-      new_attachment.save!
+      # TODO: Somehow the attachable lost inside the transaction.
+      new_attachment_reference.attachable = self
```

Comment:
I see why. We set the referenced object to be null.

This is why I say you do need to pass attachments around sometimes.